massa-pos-exports/src/pos_final_state.rs

```diff
@@ -259,7 +265,11 @@ impl PoSFinalState {
                 if !cycle_info.complete {
                     return Err(PosError::CycleUnfinished(c));
                 }
-                Hash::compute_from(&cycle_info.rng_seed.clone().into_vec())
+                let mut seed = cycle_info.rng_seed.clone().into_vec();
```

Comment:
I think we should also prepend the serialized current cycle number to the seed in order to make it really unique